### DISCUSSION TITLE
how we can accept an argument to a command

### DISCUSSION THREAD

**Discussion by lexjwa** [deployphp/deployer#2824] [Help needed]
We are working with deployer ^7.0.

We want to pass a argument to command running a specific task.

```
// Deployer command to sync .env
task('sync_env', [
    'deploy:info',
    'deploy:setup',
    'deploy:lock',
    'sync_env_file',
    'deploy:unlock',
]);

task('sync_env_file', function () {
    $env = askChoice("Which environment file would you like to use?", ['.env', '.env.dev', '.env.staging', '.env.production', '.env.worker', '.env.application'], 0);
    $env_to_sync = ($_SERVER['HOME'].'/environments/' . $env); //Path to new .env to sync for selected hosts.
    upload($env_to_sync, '{{release_or_current_path}}/');
    cd('{{release_or_current_path}}/');
    run('rm .env');
    run('cp ' . $env . ' .env');
    run('rm ' . $env);
    run('php artisan config:cache');
    run('php artisan optimize:clear');
});
```
We use this command to run task.

```
dep sync_env server_type=worker
```

And it asks to chose a file name to use, but instead of that we want to pass an argument to command and take that file name as input.

For example

```
dep sync_env  file_name_to_sync server_type=worker
``` 

Can anyone let me know if this is doable? Thanks

**Comment by antonmedv** [comment]
Yes, two options: environment cars or cli options.

**Reply by russellfeeed** [reply]
Yep, Just tried it,  use the standard PHP $argv

**Reply by lexjwa** [reply]
The file names I want to pass are environment files, for example. .env, .env.dev, .env.staging

`dep sync_env  .env.staging server_type=worker`

This is just a file name, not actual file.

**Reply by antonmedv** [reply]
Okay 👌

**Reply by nathan-io** [reply]
We need to be able to get both the value of the "filename" argument, and the value given for the `server_type` label.

**Reply by antonmedv** [reply]
Get(labels)[server type]

**Reply by nathan-io** [reply]
```
Get(labels)[server type]
```

Hi @antonmedv can you confirm the syntax on this?

---

> Yep, Just tried it, use the standard PHP $argv

@russellfeeed Any chance you could post an example? On PHP 8.0, I'm getting "Undefined variable $argv" on `var_dump($argv);`
